palindomepartitioning.py

```diff
@@ -0,0 +1,41 @@
+# Time Complexity : Incomplete
+# Space Complexity : Incomplete
+# Did this code successfully run on Leetcode : No
+# Any problem you faced while coding this : Yes , couldn't finish it. Got confused with how to add tokens to partition and indexing while backtracking
+
+# Your code here along with comments explaining your approach
+class Solution:
+    def partition(self, s: str) -> List[List[str]]:
+        
+        result = []
+        token = ""
+        self.helper(s, token,0, [], result)
+        return result
+    
+    def helper(self, s, token, index, partition, result):
+        
+        #isWorthExploring
+        if index!=0 and index <= len(s):
+            partition.append(token)
```

Comment:
at the first iteration, you add "aa" to the partition and call the helper function in the for loop
then for the second iteration, "ab" is checked if it's palindrome or not, palindrome function returns false and you pop out "aa" which is a valid palindrome partition. 
check the base condition, try to add the partitions in the base conditions, makes the recursion easier to understand. 

This was the first issue I encountered, let me know if I'm wrong.